cosmic-core/server/src/main/java/com/cloud/network/NetworkModelImpl.java

```diff
@@ -1751,10 +1751,12 @@ public boolean isVmPartOfNetwork(final long vmId, final long ntwkId) {
     @Override
     public boolean isPrivateGateway(final long ntwkId) {
         final Network network = getNetwork(ntwkId);
-        if (network.getTrafficType() != TrafficType.Guest || network.getNetworkOfferingId() != s_privateOfferingId.longValue()) {
-            return false;
+        if (network.getGuestType() != null && network.getGuestType().equals(Network.GuestType.Private)) {
```

Comment:
I'd replace the `if` expression with: `return Network.GuestType.Private.equals(network.getGuestType());` and I'd log the network type right before the `return` clause. Something like: `s_logger.debug("Network [" + network.getName() + "] is a " + network.getGuestType() + " network.");`